cmd/gateway/s3/gateway-s3.go

```diff
@@ -198,6 +198,25 @@ var defaultAWSCredProviders = []credentials.Provider{
 	&credentials.EnvMinio{},
 }
 
+var metrics = minio.NewGatewayMetrics()
+
+type statTransport struct {
+	originalTransport *http.Transport
+	metrics           *minio.GatewayMetrics
+}
+
+func (s statTransport) RoundTrip(r *http.Request) (*http.Response, error) {
+	s.metrics.IncRequests()
```

Comment:
Just added a println to see the `r.URL`.. I see that we increment even for non-s3 request uri's such as `firebase-messaging-sw.js/?location=` , `/assets/?location=`.. These occur during browser refreshes.

And also we record all the sanity calls.. A single PutObject call would hit `getBucketLocation` calls internally..  In this case, for a single put, there will be more than one request counts getting incremented.

Just need to know if these things are intentional?

Reply:
IMO `getBucketLocation` calls to S3 should be recorded, but the browser refresh should not. Let me fix this

Reply:
We should also perhaps make backend S3 requests granular just like what we did with server - this allows for a more richer metrics. They may be interested in this too, like ttfb from the backend etc.

You may have to replicate something similar to collectStats()

Reply:
okay, makes sense. Let me explore this..

Reply:
Since paths are unbounded, it causes cardinality issues in Prometheus and hence standard approach is not use paths as labels in Prometheus metrics, ref: https://github.com/prometheus/client_golang/issues/491, https://github.com/miekg/caddy-prometheus/issues/50 .